src/coreclr/jit/importercalls.cpp

```diff
@@ -1842,10 +1861,74 @@ void Compiler::impPopArgsForUnmanagedCall(GenTreeCall* call, CORINFO_SIG_INFO* s
 
     if (call->unmgdCallConv == CorInfoCallConvExtension::Thiscall)
     {
-        assert(argsToReverse);
+        assert(argsToReverse != 0);
         argsToReverse--;
     }
 
+#ifdef SWIFT_SUPPORT
+    unsigned short swiftErrorIndex = sig->numArgs;
+
+    // We are importing an unmanaged Swift call, which might require special parameter handling
+    if (call->unmgdCallConv == CorInfoCallConvExtension::Swift)
+    {
+        bool spillAllArgs = false;
+
+        // Check the signature of the Swift call for the special types
+        CORINFO_ARG_LIST_HANDLE sigArg = sig->args;
+
+        for (unsigned short argIndex = 0; argIndex < sig->numArgs;
```

Comment:
Is it possible to avoid this loop and assign context registers for Swift args in the register allocator? It should simplify the SwiftSelf argument as well, as in that case it can be treated as any other argument.

Reply:
The loop cannot be removed. We do not keep call site signature information beyond the point here, so some handling of `SwiftError` and `SwiftSelf` has to be done doing import.

There is nothing related to register handling here; registers are always assigned by LSRA. ABI information determination is done later than this but before LSRA; we have a mechanism to mark certain arguments as "well known arguments", and those arguments can get custom register assignments, so that is going to be used for `SwiftSelf`.

Modelling `SwiftError` as an argument for a longer time would be possible, but I don't see a good reason to do that since it's not a faithful representation. It would mean that we cannot keep the error enregistered. We get rid of the "argument" representation very early here and consider it as an actual definition from a value instead (modelled by the `GT_SWIFT_ERROR` node), like any other definition of a value.